src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -277,31 +278,17 @@ public CatalogObjectMetadata updateObjectMetadata(String bucketName, String name
             throw new TagsIsNotValidException(tags.get(), "tags");
         }
 
-        //synchronize project name values
-        for (KeyValueLabelMetadataEntity metadata : catalogObjectRevisionEntity.getKeyValueMetadataList()) {
-            if (metadata.getKey().equals(PROJECT_NAME)) {
-                metadata.setValue(projectName.orElse(""));
-            }
+        if (!catalogObjectRevisionEntity.getProjectName().equals(projectName.orElse("")) ||
+            !catalogObjectRevisionEntity.getTags().equals(tags.orElse(""))) {
+            buildCatalogObjectRevisionEntity("The project name or/and tags metadata are updated",
+                                             username,
+                                             projectName.orElse(""),
+                                             tags.orElse(""),
```

Comment:
I would prefer that, if the projectName or tags is absent, we take the value from the previous revision instead of setting it to empty.